mpqp/noise/noise_model.py

```diff
@@ -30,14 +31,14 @@ class NoiseModel(ABC):
 
     Raises:
         ValueError: When target list is empty, or target indices are duplicated
-            or negative. When a custom gate does not have a class attribute ``nb_qubits``.
-            When the size of the gate is higher than the number of target qubits.
+        or negative. When the size of the gate is higher than the number of target qubits.
     """
 
-    def __init__(self, targets: list[int], gates: Optional[list[type[Gate]]] = None):
-        if len(targets) == 0:
-            raise ValueError("Expected non-empty target list")
-
+    def __init__(
+        self, targets: Optional[list[int]] = None, gates: Optional[list[type[Gate]]] = None
+    ):
+        if targets is None:
+            targets = []
         if len(set(targets)) != len(targets):
             raise ValueError(f"Duplicate indices in targets: {targets}")
```

Comment:
If we allow ``targets`` to be empty in the case when we want all qubits to be targets, then we will always enter this condition of duplicate indices, no? One workaround would be to check if len(targets)!=0 first

Reply:
Indeed, my fault